src/main/scala/com/ing/wbaa/rokku/proxy/handler/namespace/NamespacesHandler.scala

```diff
@@ -0,0 +1,84 @@
+package com.ing.wbaa.rokku.proxy.handler.namespace
+
+import akka.http.scaladsl.model.HttpRequest
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.s3.model.{ AmazonS3Exception, ListObjectsV2Result }
+import com.ing.wbaa.rokku.proxy.config.NamespaceSettings
+import com.ing.wbaa.rokku.proxy.data.RequestId
+import com.ing.wbaa.rokku.proxy.handler.LoggerHandlerWithId
+import com.ing.wbaa.rokku.proxy.util.S3Utils
+
+import scala.collection.immutable.ListMap
+import scala.collection.mutable
+import scala.util.{ Failure, Success, Try }
+
+case class NamespaceName(name: String)
+
+case class BucketName(name: String)
+
+trait NamespacesHandler {
+  private val logger = new LoggerHandlerWithId
+  private val bucketCredentials: mutable.Map[BucketName, BasicAWSCredentials] = scala.collection.mutable.Map[BucketName, BasicAWSCredentials]()
```

Comment:
@kr7ysztof Probably this data structure won't be thread safe. Potential 2 requests can arrive the same time and have a race condition. From functional point of view  it won't be critical (e.g rokku will just initiate an extra find namespace request) but I don't know if we will get any runtime errors. Maybe we can use something like `Scala.collection.concurrent.Map` to have a piece of mind?

Reply:
right - I changed this to concurrent.Map